### ISSUE TITLE
dynamic resource allocation: simplify type names in API

### ISSUE THREAD

**Issue by pohly** [kubernetes/kubernetes#113771]
### What would you like to be added?

Potential changes:
- Reference -> Ref in struct names - might not be consistent with other types then?
- core.ResourceClaim -> ResourceClaimUse, ResourceClaimAttachment, etc. - TBD

This only affects Go code and thus is not an API change.

### Why is this needed?

Simpler and/or less ambiguous code.

**Comment by neolit123** [comment]
@pohly can you please label all issues that are "dynamic resource allocation*" with some SIG labels, so that they are not with "needs-sig". thanks!

**Comment by pohly** [comment]
I'll get to that now. I was a bit busy end of last week with the code freeze :sweat_smile: 

/sig node

**Comment by pohly** [comment]
I proposed `ResourceAttachment`, but @bart0sh correctly pointed out that `Claims: []api.ResourceAttachment{{Name: "my-claim"}},` in source code looks also weird. We also discussed `ResourceClaimAttachment`, but that wasn't much better. 

Ultimately we questioned whether the confusion between `core.ResourceClaim` and `resource.Claim` really is a problem (if we are careful when talking about code, it'll be clear what is meant) and therefore would like to propose that we just leave it as-is. 

The same applies to `*Reference` structs: for consistency with other such structs in the core API, let's keep the longer name instead of shortening to `*Ref`.

@thockin: agreed? If yes, then we can close this issue.
